CMakeLists.txt

```diff
@@ -108,9 +108,9 @@ if (NOT CMAKE_BUILD_TYPE)
   set ( CMAKE_BUILD_TYPE Release )
 endif()
 
-set ( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -fno-rtti -fvisibility-inlines-hidden" )
-set ( CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -DVC_DEBUG -fvisibility-inlines-hidden" )
-set ( CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3 -fPIC -fvisibility-inlines-hidden" )
+set ( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -ldl -fno-use-cxa-atexit" )
```

Comment:
Probably they were not required. I'd remove them if you agree with me.

Reply:
I remember `-fvisibility-inline-hidden` was added to resolve potential conflicts with global variable names when using libVC as external library. Not sure if that is still required again. Did you investigate before removing it?

Reply:
No, sorry, I didn't know the background which led to those edits, I just kept them untouched from marcof's fork

Reply:
might be worth setting [this variable](https://cmake.org/cmake/help/latest/variable/CMAKE_VISIBILITY_INLINES_HIDDEN.html#variable:CMAKE_VISIBILITY_INLINES_HIDDEN) to avoid exporting symbols that might create conflicts in the code using our library.

Reply:
About the other flag, my google-fu says that it might not be very useful. [Link](https://github.com/offa/stm32-eth/issues/23) In our case can you double check if we actually have a difference in the library binary (using gcc and using clang) with/without this flag?

Reply:
> About the other flag, my google-fu says that it might not be very useful. [Link](https://github.com/offa/stm32-eth/issues/23) In our case can you double check if we actually have a difference in the library binary (using gcc and using clang) with/without this flag?

The binary files differ, I'd set the cmake flag then

Reply:
> About the other flag, my google-fu says that it might not be very useful. [Link](https://github.com/offa/stm32-eth/issues/23) In our case can you double check if we actually have a difference in the library binary (using gcc and using clang) with/without this flag?

Both testing binaries and the library binary show a few differences, so I'd guess `-fno-use-cxa-atexit` is needed.

Reply:
> I remember `-fvisibility-inline-hidden` was added to resolve potential conflicts with global variable names when using libVC as external library. Not sure if that is still required again. Did you investigate before removing it?

https://cmake.org/cmake/help/latest/policy/CMP0063.html
I'd guess it is ok to remove it but I'm not 100% sure

Reply:
I would suggest, if you discover the reason, to add that information in the code as a comment